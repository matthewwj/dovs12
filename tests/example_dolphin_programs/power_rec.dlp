void printInt(i : int){
   output_string(int_to_string(i),get_stdout());
   output_string("\n",get_stdout());
}

int power(a: int, n: int) {
    if (n <= 0)
      return 1;
    var r = power(a, n / 2);
    r = r * r;
    if (n % 2 != 0)
      r = r * a;
    return r;
}


int main() {
    printInt(power(2, 4)); 
    printInt(power(6, 3)); 
    return 0;
}
