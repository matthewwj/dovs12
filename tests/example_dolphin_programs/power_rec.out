16
216
0
